src/Service.Tests/SqlTests/GraphQLMutationTests/GraphQLMutationTestBase.cs

```diff
@@ -206,6 +206,100 @@ public async Task TestStoredProcedureMutationForDeletion(string dbQueryToVerifyD
             Assert.AreEqual(updatedResult.RootElement.GetProperty("maxId").GetInt64(), 13);
         }
 
+        public async Task InsertMutationOnTableWithTriggerWithNonAutoGenPK(string dbQuery)
+        {
+            // Validate that insert operation succeeds when an insert DML trigger is enabled for a table with non-autogenerated primary key.
+            // For non-autogenerated PK fields, the values are inserted into a temporary table and then an INNER JOIN is performed
+            // based on those fields' values from the actual table and the temporary table.
+            // Trigger behavior: The trigger updates the salary to max(0,min(30,salary)).
+            // We would not need to specify the value of the PK column 'months' as it has a default value of 2.
```

Comment:
You didn't need to explain how the backend logic works. A more helpful comment that describes this test's validation would cover:
- given input `item: { id: 4, name: ""Tommy"", salary: 45 }` 
- this test proves that the selection set would return: `id: 4, months: 2, name: Tommy, salary: 30`
- Because the selection set shows `salary:30` that proves the insert trigger executed and the selection set included the value modified by the trigger.

Reply:
You can extrapolate this type of test description to the other related tests below.

Reply:
Thanks for the suggestions! I have updated the comments/summaries.